src/range_proof/mod.rs

```diff
@@ -204,27 +221,126 @@ impl RangeProof {
                 .chain(h)
                 .chain(x_sq.iter().cloned())
                 .chain(x_inv_sq.iter().cloned()),
-            iter::once(&self.A)
-                .chain(iter::once(&self.S))
+            iter::once(&self.A.decompress().ok_or(
+                "RangeProof.A is invalid Ristretto point",
```

Comment:
I think it would be cleaner not to interleave the error handling and the chaining here.